src/lib/cls.options.ts

```diff
@@ -72,27 +72,27 @@ export class ClsMiddlewareOptions {
      * Function that executes after the CLS context has been initialised.
      * It can be used to put additional variables in the CLS context.
      */
-    setup?: (cls: ClsService, req: any) => void | Promise<void>;
+    setup?: (cls: ClsService, req: any, res?: any) => void | Promise<void>;

```

Comment:
No need to mark it as optional, the `any` already takes care of that. Additionally if a third argument is specified, it will always be the `Response` and never `undefined`, so it is actually a "wrong" type signature.